### PR TITLE
Add test case verifying duplicate tag removal

### PR DESCRIPTION
**Pull request by sambible**
### Problem Statement
When podman containers were pushed to the same repo, with the same tag, duplicate tags weren't removed, leading to multiple containers with the same tag.

### Solution
This adds a test to verify the new deduplication behavior. This also moves this test, and others in the "TestPodman" class within test_docker.py into the ContainerImageManagement component, instead of the Repositories component, which should encompass these type of tests.

### Related Issues
https://issues.redhat.com/browse/SAT-38954

### PRT
```
trigger: test-robottelo
pytest: tests/foreman/api/test_docker.py -k 'test_podman_push_existing_tag'
```

### CODE REVIEW

**Comment by sambible** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/api/test_docker.py -k 'test_podman_push_existing_tag'
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 13486
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_docker.py -k test_podman_push_existing_tag --external-logging
Test Result : =========== 1 failed, 45 deselected, 5 warnings in 897.56s (0:14:57) ===========
```

**Comment by sambible** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/api/test_docker.py -k 'test_podman_push_existing_tag'
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 13493
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_docker.py -k test_podman_push_existing_tag --external-logging
Test Result : =========== 1 passed, 45 deselected, 5 warnings in 777.11s (0:12:57) ===========
```

File: tests/foreman/api/test_docker.py [hunk 1]
```diff
@@ -758,3 +758,63 @@ def test_cv_podman(
...
+            1. Using podman, pull an image..
+            2. Push that image with a specific tag.
+            3. Pull the same image with a different tag (Latest vs v44, for example).
+            4. Push the second image with the previously used tag.
+
+        :expectedresults: When the second image is pushed, the tag is changed to the newly pushed image,
+            and there is no tag on the previous image.
+
+        :Verifies: SAT-38954
+        :CaseImportance: High
```
Comment (vsedmik) [inline tests/foreman/api/test_docker.py]: We could inherit the module `CaseImportance` (it is `High` too)
```suggestion
```

File: tests/foreman/api/test_docker.py [hunk 2]
```diff
@@ -758,3 +758,63 @@ def test_cv_podman(
...
+        assert (
+            module_target_sat.execute(f'podman pull registry.fedoraproject.org/{image_2}').status
+            == 0
+        )
+        image_1_id = module_target_sat.execute(f'podman images {image_1} -q')
+        assert image_1_id
+        image_2_id = module_target_sat.execute(f'podman images {image_2} -q')
+        assert image_2_id
+        # Podman pushes require lowercase org and product labels
+        push_cmd = f'{(module_org.label)}/{(module_product.label)}/{image_tag}'.lower()
```
Comment (vsedmik) [inline tests/foreman/api/test_docker.py]: I would rather call it "distribution_path" per pulp terminology, or just "path".

File: tests/foreman/api/test_docker.py [hunk 3]
```diff
@@ -758,3 +758,63 @@ def test_cv_podman(
...
+        assert repo_data[0]['tags'][0]['name'] == 'test_tag'
+        assert not repo_data[1]['tags']
+        # Check that the appropriate message is logged
+        log_message = f"Removing 1 duplicate docker tag associations in repository '{image_name}'"
+        assert (
+            log_message
+            in module_target_sat.execute(
+                f"""grep {log_message} /var/log/foreman/production.log"""
+            ).stdout
+        )
```
Comment (vsedmik) [inline tests/foreman/api/test_docker.py]: Hmm, if I read it correctly, this just greps "Removing" word from the production.log and check that at least one of the returned lines contains the entire `log_message`.

Did you intend to put the `{log_message}` to double quotes?
```suggestion
        assert (
            module_target_sat.execute(
                f"""grep "{log_message}" /var/log/foreman/production.log"""
            ).status == 0
        )
```

File: tests/foreman/api/test_docker.py [hunk 4]
```diff
@@ -758,3 +758,63 @@ def test_cv_podman(
...
+            f'podman push --creds {creds} {image_2_id.stdout.strip()} {module_target_sat.hostname}/{push_cmd}'
+        )
+        assert result.status == 0, result.stderr
+        repo = module_product.read().repository
+        repo_data = module_target_sat.api.Repository(id=repo[0].id).docker_manifests()['results']
+        # Check that only the newly pushed repo has the tag
+        assert repo_data[0]['tags'][0]['name'] == 'test_tag'
+        assert not repo_data[1]['tags']
+        # Check that the appropriate message is logged
+        log_message = f"Removing 1 duplicate docker tag associations in repository '{image_name}'"
```
Comment (qcjames53) [inline tests/foreman/api/test_docker.py]: Nice check 👍

File: tests/foreman/api/test_docker.py [hunk 5]
```diff
@@ -758,3 +758,63 @@ def test_cv_podman(
...
+            f'podman push --creds {creds} {image_1_id.stdout.strip()} {module_target_sat.hostname}/{push_cmd}'
+        )
+        assert result.status == 0, result.stderr
+        result = module_target_sat.execute(
+            f'podman push --creds {creds} {image_2_id.stdout.strip()} {module_target_sat.hostname}/{push_cmd}'
+        )
+        assert result.status == 0, result.stderr
+        repo = module_product.read().repository
+        repo_data = module_target_sat.api.Repository(id=repo[0].id).docker_manifests()['results']
+        # Check that only the newly pushed repo has the tag
```
Comment (qcjames53) [inline tests/foreman/api/test_docker.py]:
```suggestion
        # Verify the repository has two docker manifests, one manifest has the test tag, and one manifest has no tags.
```

File: tests/foreman/api/test_docker.py [hunk 6]
```diff
@@ -605,7 +605,7 @@ def test_positive_remove_docker_repo_ccv(
 class TestPodman:
```
[the remark this answers is not in the document]
Comment (sambible) [reply tests/foreman/api/test_docker.py]: We do, it's basically the replacement for docker operations, since we don't really support that explictly anymore. "Container images" type stuff.
